x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/saml/sp/SamlServiceProviderDocument.java

```diff
@@ -247,6 +250,68 @@ public int hashCode() {
     public final AttributeNames attributeNames = new AttributeNames();
     public final Certificates certificates = new Certificates();
 
+    public SamlServiceProviderDocument() {
+    }
+
+    public SamlServiceProviderDocument(StreamInput in) throws IOException {
+        docId = in.readOptionalString();
+        name = in.readString();
+        entityId = in.readString();
+        acs = in.readString();
+        enabled = in.readBoolean();
+        created = in.readInstant();
+        lastModified = in.readInstant();
+        nameIdFormats = in.readSet(StreamInput::readString);
+        authenticationExpiryMillis = in.readBoolean() ? in.readVLong() : null;
+
+        privileges.application = in.readOptionalString();
+        privileges.resource = in.readString();
+        privileges.loginAction = in.readOptionalString();
+        privileges.groupActions = in.readMap(StreamInput::readString, StreamInput::readString);
+
+        attributeNames.principal = in.readString();
+        attributeNames.email = in.readOptionalString();
+        attributeNames.name = in.readOptionalString();
+        attributeNames.groups = in.readOptionalString();
+
+        certificates.serviceProviderSigning = in.readStringList();
+        certificates.identityProviderSigning = in.readStringList();
+        certificates.identityProviderMetadataSigning = in.readStringList();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeOptionalString(docId);
+        out.writeString(name);
+        out.writeString(entityId);
+        out.writeString(acs);
+        out.writeBoolean(enabled);
+        out.writeInstant(created);
+        out.writeInstant(lastModified);
+        out.writeCollection(nameIdFormats, StreamOutput::writeString);
+        if (authenticationExpiryMillis == null) {
```

Comment:
Good chance to create `StreamOutput#writeOptionalVLong` ,`StreamInput#readOptionalVLong` this could be used in ~15 places in our codebase

Reply:
Do you mind if I do it in a separate PR? I'm worried about adding it on a feature branch and it being harder to merge down the track.
I'd prefer to just raise it against master directly.

Reply:
sure thing

Reply:
https://github.com/elastic/elasticsearch/pull/53145